packages/stable/src/api/records/types.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 Cognite AS
+
+import type {
+  ContainerReference,
+  PropertyValueGroupV3,
+  RawPropertyValueV3,
+} from '../instances/types.gen';
+
+export type { ContainerReference, PropertyValueGroupV3, RawPropertyValueV3 };
+
+/**
+ * Property values for a container source
+ */
+export interface RecordData {
+  /**
+   * Reference to the container
+   */
+  source: ContainerReference;
+  /**
+   * Property values for the container
+   */
+  properties: PropertyValueGroupV3;
+}
+
+/**
+ * Record to write/ingest into a stream
+ */
+export interface RecordWrite {
+  /**
+   * The space that the record belongs to
+   */
+  space: string;
+  /**
+   * External ID of the record
+   */
+  externalId: string;
+  /**
+   * List of source properties to write
+   */
+  sources: RecordData[];
+}
+
+/**
+ * Source selector for specifying which container properties to return
+ */
+export interface SourceSelector {
+  /**
+   * Reference to the container
+   */
+  source: ContainerReference;
+  /**
+   * Properties to return for the specified container
+   */
+  properties: string[];
+}
+
+/**
+ * Last updated time filter for records
+ */
+export interface LastUpdatedTimeFilter {
+  /**
+   * Greater than or equal to
+   */
+  gte?: string | number;
```

Comment:
Nit: would be great to use a type alias for string called "LastUpdatedTimeRangeBound" or something that explains the spec for the string input here 🙂